src/integration/pages/homePage/portuguese/canonical.test.js

```diff
@@ -0,0 +1,113 @@
+      const scrollLeftButton = carousel.querySelector(
+        '[data-testid="pv-scroll-left"]',
+      );
+      const scrollRightButton = carousel.querySelector(
+        '[data-testid="pv-scroll-right"]',
+      );
+      expect(scrollLeftButton).toBeInTheDocument();
+      expect(scrollRightButton).toBeInTheDocument();
+      // cannot test the scroll functionality in a JSDOM environment with no rendering or actual scrolling
+      // also cannot test whether these appear or disappear due to screen width, as although they are in the document,
+      // these tests always sees them as not visible, as well as always disabled
+    });
+  });
+
+  it('should render each video promo item with a button containing the correct text', () => {
+    const carousels = document.querySelectorAll(
+      '[data-testid="portrait-video-carousel"] ul[data-testid="pv-carousel"]',
+    );
+
+    carousels.forEach((carousel, carouselIndex) => {
+      const promoButtons = carousel.querySelectorAll(
+        '[data-testid="promo-button"]',
+      );
+      const videoItems =
+        portraitVideoCurations[carouselIndex]?.portraitVideo?.items || [];
+      promoButtons.forEach((button, buttonIndex) => {
+        const textContents = button.querySelector(
+          '[data-testid="text-contents"]',
+        );
+        expect(textContents).toBeInTheDocument();
+        const expectedTitle = videoItems[buttonIndex]?.headlines?.promoHeadline;
+
+        expect(textContents?.textContent).toContain(expectedTitle);
+      });
+    });
+  });
+
+  it('should render each video promo item with a duration element', () => {
+    const carousels = document.querySelectorAll(
+      '[data-testid="portrait-video-carousel"] ul[data-testid="pv-carousel"]',
+    );
+    carousels.forEach(carousel => {
+      const promoButtons = carousel.querySelectorAll(
+        '[data-testid="promo-button"]',
+      );
+      promoButtons.forEach(button => {
+        // Check that the button contains a duration element
+        const duration = button.querySelector('time > span');
+        expect(duration).toBeInTheDocument();
+        expect(duration.textContent).toMatch(/\d{2}:\d{2}/);
+      });
+    });
+  });
+  it('should render each video promo item with an image', () => {
+    const carousels = document.querySelectorAll(
+      '[data-testid="portrait-video-carousel"] ul[data-testid="pv-carousel"]',
+    );
+
+    carousels.forEach((carousel, carouselIndex) => {
+      const promoItems = carousel.querySelectorAll('li');
+      const videoItems =
+        portraitVideoCurations[carouselIndex]?.portraitVideo?.items || [];
+
+      promoItems.forEach((item, itemIndex) => {
+        const image = item.querySelector('img');
+        expect(image).toBeInTheDocument();
+
+        const expectedImages = videoItems[itemIndex]?.images || [];
+
+        const expectedAlt = expectedImages[0].altText;
+        expect(image?.getAttribute('alt')).toBe(expectedAlt);
```

Comment:
Use array destructuring
```suggestion
        const [expectedImage] = videoItems[itemIndex]?.images || [];

        const { altText } = expectedImage;
        expect(image?.getAttribute('alt')).toBe(altText);
```